server/src/main/java/org/apache/cassandra/sidecar/server/MainModule.java

```diff
@@ -162,7 +164,25 @@ public Vertx vertx(SidecarConfiguration sidecarConfiguration, MetricRegistryFact
                                         // Monitor all V1 endpoints.
                                         // Additional filtering is done by configuring yaml fields 'metrics.include|exclude'
                                         .addMonitoredHttpServerRoute(serverRouteMatch);
-        return Vertx.vertx(new VertxOptions().setMetricsOptions(dropwizardMetricsOptions));
+
+        FileSystemOptions fsOptions;
+        VertxFilesystemOptionsConfiguration configuredFSOptions = sidecarConfiguration.serviceConfiguration()
+                                                                                      .vertxFilesystemOptionsConfiguration();
```

Comment:
Can `vertxFilesystemOptionsConfiguration` return null? It does not according to the implementation.
If it can, it would be good to add `@Nullable` to the signature.